src/pages/Home.js

```diff
@@ -1,35 +1,26 @@
-import React, { useEffect, useState } from 'react';
+import React, { useEffect } from 'react';
 import './Home.css';
 import BlogThumbnailContent from './components/BlogThumbnailContent';
-import { getArticles } from '../helpers/axiosConfig';
-import { Box, Card } from '@mui/material';
+
+import { Box, Card, CircularProgress } from '@mui/material';
 import { Link } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { getAllArticlesRedux } from '../features/articleSlice';
+import { getArticles } from '../features/selectors';
 
 const Home = () => {
-  let [allArticles, setAllArticles] = useState([]);
+  const dispatch = useDispatch();
+  const { articles, loadingArticles } = useSelector(getArticles);
+
   useEffect(() => {
-    getArticles().then((res) => {
-      let articlesData = res.data;
-      const normalizedArticles = articlesData.map((singleElement) => {
-        const { id, title, summary, content, author } = singleElement;
-        return {
-          id,
-          title,
-          summary,
-          content,
-          authorFirstName: author.firstName,
-          authorLastName: author.lastName,
-          image: true,
-        };
-      });
-      setAllArticles(normalizedArticles);
-    });
-  }, []);
+    dispatch(getAllArticlesRedux());
+  }, [dispatch]);
 
-  if (allArticles.length === 0) {
+  if (articles.length === 0) {
     return (
       <div className="empty-array">
         <h2>We don't have any posted articles yet</h2>
+        {loadingArticles && <CircularProgress />}
       </div>
     );
   }
```

Comment:
This can be separated to make it more readable. First check if articles are loaded (returning`CircularProgress`) and when you already have data, you can check if there are any posted articles. Something like this: 
```suggestion
  if (loadingArticles) {
    return <CircularProgress />
  }

  if (!articles.length) {
    return (
      <div className="empty-array">
        <h2>We don't have any posted articles yet</h2>
      </div>
    );
  }
```

Reply:
repaired based on the hint